packages/sources/react/src/components/structure/VtmnAccordion/VtmnAccordion.tsx

```diff
@@ -36,8 +46,22 @@ export const VtmnAccordion = ({
   open,
   children,
   className,
+  onOpen,
+  onClose,
   ...props
 }: VtmnAccordionProps) => {
+  const ref = React.useRef<HTMLDetailsElement>(null);
+  const toggleAccordion = React.useCallback(
+    (e: React.MouseEvent): void => {
+      const element = ref.current;
+      if (element) {
+        if (element.open && onClose) onClose(e);
+        if (!element.open && onOpen) onOpen(e);
+      }
+    },
+    [onOpen, onClose],
+  );
+
```

Comment:
Wouldn't it be a more "react way" (and closer to what you suggest for Svelte) to avoid the imperative ref use here and go with a full controlled component with an inner state. For instance :
```
export const VtmnAccordion = ({
  ...
  ...props
}: VtmnAccordionProps) => {
  const [isOpen, setIsOpen] = React.useState(open);
  const toggle = () => setIsOpen((prevOpen) => !prevOpen);
  React.useEffect(() => {
    if (isOpen && onClose) onClose();
    if (!isOpen && onOpen) onOpen();
  }, [isOpen]);
  return (
    <details
      ...
      open={isOpen}
      onClick={toggle}
      {...props}
    >
      ...
    </details>
  );
};
```

What do you think of it ?

Reply:
ok

Reply:
You don't want to pass the whole Event on ? Might be easier to read and even to use in future cases. You could get the event on your side when you consume the component, and then destructure it to only use target